external/storm-kafka/src/jvm/storm/kafka/PartitionManager.java

```diff
@@ -137,8 +137,14 @@ public EmitState next(SpoutOutputCollector collector) {
             }
             Iterable<List<Object>> tups = KafkaUtils.generateTuples(_spoutConfig, toEmit.msg);
             if (tups != null) {
-                for (List<Object> tup : tups) {
-                    collector.emit(tup, new KafkaMessageId(_partition, toEmit.offset));
+		if(_spoutConfig.topicAsStreamId) {
+	            for (List<Object> tup : tups) {
```

Comment:
@harshach 
@rohanag12's approach reduces execution of if statement to just once.
(And I'm not sure compiler optimization takes care of it.)
It may hurt readability slightly by code duplication, but I think it is small enough.